hoomd/mpcd/CollisionMethod.cc

```diff
@@ -48,15 +48,284 @@ void mpcd::CollisionMethod::collide(uint64_t timestep)
     // sync the embedded group
     m_cl->setEmbeddedGroup(m_embed_group);
 
+    // check for collision warnings
+    checkCollisionWarnings(timestep);
+
     // set random grid shift
     m_cl->drawGridShift(timestep);
 
     // update cell list
     m_cl->compute(timestep);
 
+    // create auxillary array for rigid bodies
+    if (m_rigid_bodies)
+        {
+        // resize initial velocity array
+        const unsigned int num_group = m_embed_group->getNumMembers();
+        if (num_group > m_initial_velocity.getNumElements())
+            {
+            GPUArray<Scalar4> initial_velocity(num_group, m_exec_conf);
+            m_initial_velocity.swap(initial_velocity);
+            }
+
+        const unsigned int num_total = m_pdata->getN();
+        if (num_total > m_linmom_accum.getNumElements())
+            {
+            GPUArray<Scalar3> linmom_accum(num_total, m_exec_conf);
+            m_linmom_accum.swap(linmom_accum);
+            GPUArray<Scalar3> angmom_accum(num_total, m_exec_conf);
+            m_angmom_accum.swap(angmom_accum);
+            }
+        else
+            {
+            m_linmom_accum.zeroFill();
+            m_angmom_accum.zeroFill();
+            }
```

Comment:
This should be done in `accumulateRigidBodyMomenta` since that function should not assume someone has zeroed its accumulators before it uses them.
```suggestion
```
FYI: putting these in an `else` branch would also be problematic because it means that the memory would not get zeroed if the arrays needed to be reallocation. Memory is not in a defined state after allocation.